neptune-core/src/application/rpc/server.rs

```diff
@@ -4163,9 +4179,96 @@ impl RPC for NeptuneRPCServer {
             .map(|tx| &tx.kernel)
             .cloned())
     }
+
+    // Crashes if `tx_ix` out of its bound. https://github.com/Neptune-Crypto/neptune-core/issues/816
+    async fn prove_transfer(
+        self,
+        context: ::tarpc::context::Context,
+        token: auth::Token,
+        tx_ix: u64,
+        utxo_ix: usize,
+        block: Digest,
+    ) -> RpcResult<(Claim, NeptuneProof)> {
```

Comment:
I'm thinking that this function implementation should be a simple wrapper around another function that lives in the API module. Then that other function can be called also by the JSON-RPC server, as well as tested independently of any RPC servers.